flink-core/src/test/java/org/apache/flink/api/java/typeutils/runtime/PojoSerializerTest.java

```diff
@@ -606,6 +665,45 @@ public void testSerializerSerializationFailureResilience() throws Exception {
 		verifyPojoSerializerConfigSnapshotWithEmptyNestedSerializers(config, deserializedConfig);
 	}
 
+	@SuppressWarnings("unchecked")
+	@Test
+	public void testNonDuplicatedStatelessSerializersInConfigSnapshot() throws IOException {
+		PojoSerializer<TestUserClassWithSameFieldTypes> pojoSerializer = (PojoSerializer<TestUserClassWithSameFieldTypes>)
+			TypeExtractor.getForClass(TestUserClassWithSameFieldTypes.class).createSerializer(new ExecutionConfig());
+
+		// snapshot configuration and serialize to bytes
+		PojoSerializer.PojoSerializerConfigSnapshot<TestUserClassWithSameFieldTypes> config = pojoSerializer.snapshotConfiguration();
+		byte[] serializedConfig;
+		try (
+			ByteArrayOutputStream out = new ByteArrayOutputStream()) {
```

Comment:
You could use Flink's `ByteArrayOutputStreamWithPos`, which can be more efficient because it does not use `synchronized` and you can access the write position and the internal array to immediately write it to the stream instead of creating a copy in the `out.toByteArray()` part.